lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -2033,6 +2064,7 @@ namespace Js
         SparseArraySegmentBase *seg, *nextSeg, *prevSeg = nullptr;
         for (seg = fArray->head; seg; seg = nextSeg)
         {
+            bool isInlineSegment = JavascriptArray::IsInlineSegment(seg, fArray);
```

Comment:
>isInlineSegment [](start = 17, length = 15)

Have isInlineSegment as false and move isInlineSegment = JavascriptArray::IsInlineSegment(seg, fArray); in if part where we are allocating newSeg that way when we call ClearElements we need to check only isInlineSegment